AK/Time.cpp

```diff
@@ -106,49 +89,56 @@ i64 Time::to_truncated_seconds() const
 {
     VERIFY(m_nanoseconds < 1'000'000'000);
     if (m_seconds < 0 && m_nanoseconds) {
-        Checked<i64> seconds(m_seconds);
-        seconds++;
-        return seconds.has_overflow() ? 0x7fff'ffff'ffff'ffffLL : seconds.value();
+        // Since m_seconds is negative, adding 1 can't possibly overflow
+        return m_seconds + 1;
     }
     return m_seconds;
 }
 i64 Time::to_truncated_milliseconds() const
 {
     VERIFY(m_nanoseconds < 1'000'000'000);
-    Checked<i64> milliseconds(m_seconds);
+    Checked<i64> milliseconds((m_seconds < 0) ? m_seconds + 1 : m_seconds);
     milliseconds *= 1'000;
     if (!milliseconds.has_overflow()) {
         u32 add_ms = (u32)(m_nanoseconds / 1'000'000);
-        if (add_ms) {
+        if (add_ms)
             milliseconds += add_ms;
-            if (m_seconds < 0 && m_nanoseconds % 1'000'000 != 0)
+        if (!milliseconds.has_overflow() && m_seconds < 0) {
+            if (m_nanoseconds % 1'000'000 != 0)
                 milliseconds++;
-            if (!milliseconds.has_overflow())
-                return milliseconds.value();
-        } else {
-            return milliseconds.value();
+            if (!milliseconds.has_overflow()) {
+                // we only multiplied by one second less before to avoid an
+                // overflow that may not necessarily happen depending on add_ms
+                milliseconds -= 1'000;
+            }
         }
+        if (!milliseconds.has_overflow())
+            return milliseconds.value();
     }
     return m_seconds < 0 ? -0x8000'0000'0000'0000LL : 0x7fff'ffff'ffff'ffffLL;
 }
 i64 Time::to_truncated_microseconds() const
 {
     VERIFY(m_nanoseconds < 1'000'000'000);
-    Checked<i64> microseconds(m_seconds);
+    Checked<i64> microseconds((m_seconds < 0) ? m_seconds + 1 : m_seconds);
     microseconds *= 1'000'000;
     if (!microseconds.has_overflow()) {
         u32 add_us = (u32)(m_nanoseconds / 1'000);
-        if (add_us) {
+        if (add_us)
             microseconds += add_us;
-            if (m_seconds < 0 && m_nanoseconds % 1'000 != 0)
+        if (!microseconds.has_overflow() && m_seconds < 0) {
+            if (m_nanoseconds % 1'000 != 0)
                 microseconds++;
-            if (!microseconds.has_overflow())
-                return microseconds.value();
-        } else {
-            return microseconds.value();
+            if (!microseconds.has_overflow()) {
+                // we only multiplied by one second less before to avoid an
+                // overflow that may not necessarily happen depending on add_ms
+                microseconds -= 1'000'000;
+            }
         }
+        if (!microseconds.has_overflow())
+            return microseconds.value();
     }
-    return m_seconds < 0 ? -0x8000'0000'0000'0000LL : 0x7fff'ffff'ffff'ffffLL;
+    return m_seconds < 0 ? -0x8000'0000'0000LL : 0x7fff'ffff'ffffLL;
```

Comment:
This is too specific to be a typo, but too meaningless to be intentional. What happened here? :D

The correct values are `I64_MIN` and `I64_MAX`, respectively. No need to divide by 65536. Int saturation is int saturation.